pkg/engine/context/context_test.go

```diff
@@ -123,3 +124,77 @@ func Test_addResourceAndUserContext(t *testing.T) {
 		t.Error("expected result does not match")
 	}
 }
+
+func TestAddVariable(t *testing.T) {
+	conf := config.NewDefaultConfiguration(false)
+	jp := jmespath.New(conf)
+	ctx := NewContext(jp)
+
+	tests := []struct {
+		name     string
+		key      string
+		value    interface{}
+		wantErr  bool
+		expected interface{}
+	}{
+		{
+			name:    "Simple variable",
+			key:     "simpleKey",
+			value:   "simpleValue",
+			wantErr: false,
+		},
+		{
+			name:    "Nested variable",
+			key:     "nested.key",
+			value:   123,
+			wantErr: false,
+		},
+		{
+			name:    "Invalid key format",
+			key:     "invalid,key",
+			value:   "someValue",
+			wantErr: false,
+		},
+		{
+			name:    "Complex nested variable",
+			key:     "complex.nested.key",
+			value:   map[string]interface{}{"innerKey": "innerValue"},
+			wantErr: false,
+		},
+		{
+			name:    "Array value",
+			key:     "arrayKey",
+			value:   []int{1, 2, 3},
+			wantErr: false,
+		},
+		{
+			name:    "Boolean value",
+			key:     "boolKey",
+			value:   true,
+			wantErr: false,
+		},
+		{
+			name:    "Empty key",
+			key:     "",
+			value:   "someValue",
+			wantErr: true,
+		},
+		{
+			name:    "Nil value",
+			key:     "nilKey",
+			value:   nil,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ctx.AddVariable(tt.key, tt.value)
```

Comment:
You seem to only be testing here if the function did not emit an error. I think it would be valuable to also test if the Variable is actually added to the context as expected.